src/sema/yul/switch.rs

```diff
@@ -42,6 +45,32 @@ pub(crate) fn resolve_switch(
         next_reachable |= block_reachable;
     }
 
+    let mut conditions: HashSet<BigInt> = HashSet::new();
+    for item in &case_blocks {
+        let big_int = match &item.condition {
+            YulExpression::BoolLiteral(_, value, _) => {
+                if *value {
+                    BigInt::one()
+                } else {
+                    BigInt::zero()
+                }
+            }
+            YulExpression::NumberLiteral(_, value, _) => value.clone(),
+            YulExpression::StringLiteral(_, value, _) => BigInt::from_bytes_be(Sign::Plus, value),
+            _ => unreachable!("Switch condition should be a literal"),
+        };
+
+        if conditions.contains(&big_int) {
+            ns.diagnostics.push(Diagnostic::error(
+                item.loc,
+                "duplicate case for switch".to_string(),
+            ));
```

Comment:
Please use `Diagnostic::error_with_note()` and provide the location of other entry which is duplicate.

Reply:
Also use the location of the literal, not the entire case including block